src/main/scala/strawman/collection/mutable/Iterator.scala

```diff
@@ -4,9 +4,10 @@ import scala.{Boolean, Int, Unit, Nothing, NoSuchElementException}
 import strawman.collection.{IndexedView, IterableOnce}
 
 /** A core Iterator class */
-trait Iterator[+A] { self =>
+trait Iterator[+A] extends IterableOnce[A] { self =>
```

Comment:
I’m still afraid that this could lead to very surprising behaviors. Consider for instance the following program:

~~~ scala
def f(xs: TraversableOnce[Int]): Unit = {
  println(xs.iterator().next())
  println(xs.iterator().next())
}
~~~

We basically create a new iterator and print its first element, twice in a row. If I run this program I would expect to see the same value printed twice. However, if I pass an `Iterator[Int]` as a parameter to `f`, it will print the first element and then the second element. That’s because `iterator()`’s implementation in `Iterator` returns `this` instead of a fresh iterator.